lib/Sema/ConstraintSystem.cpp

```diff
@@ -701,8 +701,11 @@ ConstraintLocator *ConstraintSystem::getCalleeLocator(
       auto *fnExpr = applyExpr->getFn();
 
       // Handle special cases for applies of non-function types.
-      if (auto *loc = getSpecialFnCalleeLoc(getType(fnExpr)))
-        return loc;
+      if (hasType(fnExpr)) {
+        if (auto *loc = getSpecialFnCalleeLoc(getType(fnExpr))) {
```

Comment:
This doesn't seem right, `getType` is a function being passed into `getCalleeLocator` whereas `hasType` is the `ConstraintSystem` member; you'd need to plumb through a separate `hasType` function. That being said I'm curious why the type wouldn't be set here 🤔

Reply:
@ahoppen I dug into this a bit and discovered it's because we're re-creating the ExprPattern along different solver paths, put up #74387 to fix that which should mean we don't have to do this.

Reply:
Thank you! ⭐